File: app/src/main/java/com/kelsos/mbrc/di/providers/ContextProvider.java
package com.kelsos.mbrc.di.providers;

import android.app.Application;
import android.content.Context;
import javax.inject.Inject;
import javax.inject.Provider;

public class ContextProvider implements Provider<Context> {
  private Application application;

  @Inject
  public ContextProvider(Application application) {
    this.application = application;
  }

  @Override
  public Context get() {
    return application.getApplicationContext();
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/di/providers/ContextProvider.kt
package com.kelsos.mbrc.di.providers

import android.app.Application
import android.content.Context
import javax.inject.Inject
import javax.inject.Provider

class ContextProvider
@Inject
constructor(private val application: Application) : Provider<Context> {

  override fun get(): Context {
    return application.applicationContext
  }
}
